src/components/EmbeddedWalletModal/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button, Modal } from "@stellar/design-system";
+
+import "./styles.scss";
+
+interface EmbeddedWalletModalProps {
+  visible: boolean;
+  onClose: () => void;
+  title: string;
+  description?: string;
+  contentAlign?: "center" | "left";
+  primaryActionLabel?: string;
+  onPrimaryAction?: () => void;
+  isPrimaryActionLoading?: boolean;
+  isPrimaryActionDisabled?: boolean;
+}
+
+export const EmbeddedWalletModal = ({
+  visible,
+  onClose,
+  title,
+  description,
+  contentAlign = "center",
+  primaryActionLabel,
+  onPrimaryAction,
+  isPrimaryActionLoading = false,
+  isPrimaryActionDisabled = false,
+}: EmbeddedWalletModalProps) => {
+  const handleClose = () => {
+    onClose();
+  };
+
+  const contentAlignClass = `EmbeddedWalletModal__content--${contentAlign}`;
+  const headingAlignClass = `EmbeddedWalletModal__heading--${contentAlign}`;
+
+  return (
+    <Modal parentId="embedded-wallet-modal-root" visible={visible} onClose={handleClose}>
+      <Modal.Heading>
+        <div className={`EmbeddedWalletModal__heading ${headingAlignClass}`}>
```

Comment:
It might be cleaner to use `data-` attributes instead of classes for styling like this.

```suggestion
        <div className="EmbeddedWalletModal__heading" data-align={contentAlign}>
```